### PR TITLE
Feature/iwtf 2961 analytics permission banner

### PR DESCRIPTION
**Pull request by ScottDormand96**
https://eaflood.atlassian.net/browse/IWTF-2961

adding functionality so able to accept/reject tracking analytics, removing the need  for jquery

### CODE REVIEW

File: packages/gafl-webapp-service/src/session-cache/add-permission.js [hunk 1]
```diff
@@ -24,11 +25,13 @@ export default async request => {
   page.permissions.push({})
   status.permissions.push({})
   addressLookup.permissions.push({})
+  analytics.permissions.push({})
```
Comment (jaucourt) [inline packages/gafl-webapp-service/src/session-cache/add-permission.js:28]: I don't think this is needed

**Review by jaucourt — CHANGES_REQUESTED** [review]

File: packages/gafl-webapp-service/src/plugins.js [hunk 2]
```diff
@@ -69,6 +70,15 @@ const initialiseHapiGapiPlugin = () => {
     plugin: HapiGapi,
     options: {
       propertySettings: hapiGapiPropertySettings,
+      trackAnalytics: async request => {
+        const canTrack = await checkAnalytics(request)
+        if (canTrack === true) {
+          console.log('session is tracked')
+        } else {
+          console.log('session is not tracked')
+        }
```
[the remark this answers is not in the document]
Comment (ScottDormand96) [reply packages/gafl-webapp-service/src/plugins.js:79]: Should we not still track if logged as part of acceptance criteria?

File: packages/gafl-webapp-service/src/session-cache/add-permission.js (hunk 1 above)
Comment (irisfaraway) [reply packages/gafl-webapp-service/src/session-cache/add-permission.js:28]: Have we decided if this is being changed?

File: packages/gafl-webapp-service/src/plugins.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (irisfaraway) [reply packages/gafl-webapp-service/src/plugins.js:79]: AC does say this should be disabled in prod to avoid noise

**Review by irisfaraway — COMMENTED** [review]
This is looking good so far, but it looks like there are still some unresolved questions and feedback as well as the broken test issue, so don't think it's ready for PR approval yet

File: packages/gafl-webapp-service/src/locales/en.json [hunk 3]
```diff
@@ -104,6 +104,14 @@
   "age_junior": "Junior",
   "age_senior_concession": "Over 65",
   "age_senior": "Senior",
+  "analytics_banner_acceptance": "You have accepted analytics, thank you.",
+  "analytics_banner_preamble_1": "Please help us to improve this service by allowing us to analyse how you progress through it. This is completely anonymous, we won't record any personal information or use any cookies to do this, we simply log the pages you visit. ",
```
Comment (irisfaraway) [inline packages/gafl-webapp-service/src/locales/en.json:108]:
```suggestion
  "analytics_banner_preamble_1": "Please help us to improve this service by allowing us to analyse how you progress through it. This is completely anonymous. We will not record any personal information or use any cookies to do this – we simply log the pages you visit.",
```

File: packages/gafl-webapp-service/src/locales/en.json [hunk 4]
```diff
@@ -104,6 +104,14 @@
...
+  "analytics_banner_preamble_2": "This website only stores the cookies that are essential to make it work, Read more about ",
```
Comment (irisfaraway) [inline packages/gafl-webapp-service/src/locales/en.json:109]:
```suggestion
  "analytics_banner_preamble_2": "This website only stores the cookies that are essential to make it work. Read more about ",
```

File: packages/gafl-webapp-service/src/locales/cy.json [hunk 5]
```diff
@@ -104,6 +104,14 @@
   "age_junior": "Iau",
   "age_senior_concession": "Over 65",
   "age_senior": "Senior",
+  "analytics_banner_acceptance": "You have accepted analytics, thank you.",
+  "analytics_banner_preamble_1": "Please help us to improve this service by allowing us to analyse how you progress through it. This is completely anonymous, we won't record any personal information or use any cookies to do this, we simply log the pages you visit. ",
```
Comment (irisfaraway) [inline packages/gafl-webapp-service/src/locales/cy.json:108]:
```suggestion
  "analytics_banner_preamble_1": "Please help us to improve this service by allowing us to analyse how you progress through it. This is completely anonymous. We will not record any personal information or use any cookies to do this – we simply log the pages you visit.",
```

File: packages/gafl-webapp-service/src/locales/cy.json [hunk 6]
```diff
@@ -104,6 +104,14 @@
...
+  "analytics_banner_preamble_2": "This website only stores the cookies that are essential to make it work, Read more about ",
```
Comment (irisfaraway) [inline packages/gafl-webapp-service/src/locales/cy.json:109]:
```suggestion
  "analytics_banner_preamble_2": "This website only stores the cookies that are essential to make it work. Read more about ",
```

**Review by irisfaraway — CHANGES_REQUESTED** [review]
I discussed the wording of this with Melinda and agreed to make some very minor changes to the text to fix punctuation and also remove the negative contraction. Confirmation on Jira too.

File: packages/gafl-webapp-service/src/session-cache/add-permission.js (hunk 1 above)
Comment (jaucourt) [reply packages/gafl-webapp-service/src/session-cache/add-permission.js]: Yeah, think it needs to go

File: packages/gafl-webapp-service/src/handlers/page-handler.js [hunk 7]
```diff
@@ -105,6 +108,14 @@ export default (path, view, completion, getData) => ({
     pageData.mssgs = request.i18n.getCatalog()
     pageData.altLang = request.i18n.getLocales().filter(locale => locale !== request.i18n.getLocale())
     pageData.backRef = await getBackReference(request, view)
+    pageData.uri = { ...(pageData.uri || {}), analyticsFormAction: PROCESS_ANALYTICS_PREFERENCES.uri }
+
+    const analytics = await request.cache().helpers.analytics.get()
+
+    pageData.analyticsMessageDisplayed = analytics[ANALYTICS.seenMessage]
+    pageData.analyticsSelected = analytics[ANALYTICS.selected]
+    pageData.acceptedTracking = analytics[ANALYTICS.acceptTracking]
```
[the remark this answers is not in the document]
Comment (ScottDormand96) [reply packages/gafl-webapp-service/src/handlers/page-handler.js:117]: I added snapshot so tests all values set

**Review by irisfaraway — APPROVED** [review]

**Pull request merged**